iroh-net/src/magicsock/endpoint.rs

```diff
@@ -1018,6 +1043,53 @@ pub(super) struct PeerMap {
 }
 
 impl PeerMap {
+    /// Get the known peer addresses stored in the map. Peers with empty addressing information are
+    /// filtered out.
+    pub fn known_peer_addresses(&self) -> impl Iterator<Item = PeerAddr> + '_ {
+        self.by_id.values().filter_map(|endpoint| {
+            let peer_addr = endpoint.peer_addr();
+            (!peer_addr.info.is_empty()).then_some(peer_addr)
+        })
+    }
+
+    /// Create a new [`PeerMap`] from data stored in `path`.
+    pub fn load_from_file(
+        path: impl AsRef<Path>,
+        msock_sender: mpsc::Sender<ActorMessage>,
+    ) -> anyhow::Result<Self> {
+        let mut me = PeerMap::default();
+        let contents = std::fs::read(path)?;
```

Comment:
We could read in chunks here. However the file would need to get *really* big for this to be an issue. A `PeerAddr` will take less than 100 bytes even with a few socket addresses. So if you store 10.000 peers that'd do a ~1MB allocation here.
I think it's fine to keep as-is now, and we could move to chunked reading easily in a follow-up.